File: snips/leetcode/easy/771-jewels-and-stones.go
package main

// https://leetcode.com/problems/jewels-and-stones/

import (
	"fmt"
)

// Runtime: 0 ms, faster than 100.00% of Go online submissions for Jewels and Stones.
// Memory Usage: 1.1 MB, less than 62.50% of Go online submissions for Jewels and Stones.

func numJewelsInStones(J string, S string) int {
	chMap := make(map[int32]bool)
	for _, ch := range J {
		chMap[ch] = true
	}

	cnt := 0
	for _, ch := range S {
		//if val, ok := chMap[ch]; ok {
		//	fmt.Println(val)
		if chMap[ch] {
			cnt++
		}
	}

	return cnt
}

func main() {
	J := "aA"
	S := "aAAbbbb"
	fmt.Println(J)
	fmt.Println(S)
	fmt.Println(numJewelsInStones(J, S))
}

File: src-oth/leetcode/easy/771-jewels-and-stones.php
<?php

// https://leetcode.com/problems/jewels-and-stones/

// Runtime: 12 ms, faster than 100.00% of PHP online submissions for Jewels and Stones.

class Solution {
    function numJewelsInStones($J, $S) {
        $map = [];
        $jLen = strlen($J);
        for ($i = 0; $i < $jLen; ++$i) {
            $map[$J[$i]] = true;
        }

        $cnt = 0;
        $sLen = strlen($S);
        for ($i = 0; $i < $sLen; ++$i) {
            if ($map[$S[$i]] ?? false) {
                ++$cnt;
            }
        }
        return $cnt;
    }
}

echo $J = "aA";
echo "\n";
echo $S = "aAAbbbb";
echo "\n";
echo (new Solution())->numJewelsInStones($J, $S);
echo "\n";
